core/http-auth/src/main/java/software/amazon/awssdk/http/auth/internal/DefaultAwsV4HttpSigner.java

```
@@ -72,157 +75,225 @@ public SyncSignedRequest sign(SyncSignRequest<? extends AwsCredentialsIdentity>
+            .payload(request.payload().orElse(null))
+            .build();
     }
 
-    private SdkHttpRequest.Builder doSign(SignRequest<?, ? extends AwsCredentialsIdentity> request,
+    /**
+     * Using a {@link SignRequest} and a {@link ContentChecksum}, perform all the necessary steps to
+     * create a signed request according to the SigV4 signing documentation:
+     * <p>
+     * https://docs.aws.amazon.com/IAM/latest/UserGuide/create-signed-request.html
+     */
+    protected SdkHttpRequest.Builder sign(SignRequest<?, ? extends AwsCredentialsIdentity> signRequest,
                                           ContentChecksum contentChecksum) {
-        SdkHttpRequest.Builder requestBuilder = request.request().toBuilder();
 
-        Boolean doubleUrlEncode = validatedProperty(request, DOUBLE_URL_ENCODE, true);
-        Boolean normalizePath = validatedProperty(request, NORMALIZE_PATH, true);
-        String checksumHeaderName = request.property(CHECKSUM_HEADER_NAME);
-        Instant requestSigningInstant = validatedProperty(request, REQUEST_SIGNING_INSTANT);
-        String formattedRequestSigningDate = formatDateStamp(requestSigningInstant);
-        String formattedRequestSigningDateTime = formatTimestamp(requestSigningInstant);
-        String regionName = validatedProperty(request, REGION_NAME);
-        String serviceSigningName = validatedProperty(request, SERVICE_SIGNING_NAME);
-        String scope = buildScope(formattedRequestSigningDate, serviceSigningName, regionName);
-
-        AwsCredentialsIdentity sanitizedCredentials = sanitizeCredentials(request.identity());
-        if (sanitizedCredentials instanceof AwsSessionCredentialsIdentity) {
-            addSessionCredentials(requestBuilder, (AwsSessionCredentialsIdentity) sanitizedCredentials);
+        SdkHttpRequest.Builder requestBuilder = signRequest.request().toBuilder();
+        Instant requestSigningInstant = validatedProperty(signRequest, SIGNING_CLOCK).instant();
```

nit: move this after required region/signing name, so it can fail fast.